pkg/service/types.go

```diff
@@ -56,7 +57,7 @@ func (ms MeshService) FQDN() string {
 	if ms.ClusterDomain == "" {
 		ms.ClusterDomain = constants.LocalDomain
 	}
-	return strings.Join([]string{ms.Name, ms.Namespace, ms.ClusterDomain.String()}, ".")
+	return fmt.Sprintf("%s%s%s%s%s", ms.Name, fqdnSeparator, ms.Namespace, fqdnSeparator, ms.ClusterDomain)
```

Comment:
To really simplify this to avoid repeated usage of `fqdnSeparator` which is pretty implicit in using a period.
```suggestion
	return fmt.Sprintf("%s.%s.%s", ms.Name, ms.Namespace, fqdnSeparator, ms.ClusterDomain)
```

Reply:
Reads much better! Thanks!